backend/persistent/QueryScheduler.js

```diff
@@ -119,7 +121,69 @@ class QueryScheduler {
         Object.keys(this.queryJobs).forEach(this.clearQuery);
     }
 
-    queryAndUpdateGrid (fid, uids, queryString, connectionId, requestor) {
+    queryAndCreateGrid(filename, query, connectionId, requestor) {
+        const {username, apiKey, accessToken} = getCredentials(requestor);
+        let startTime;
+
+        // Check if the user even exists
+        if (!username || !(apiKey || accessToken)) {
```

Comment:
We can simplify it:
```
if !(username || apiKey || accessToken) {
```

Reply:
tricky, but not equivalent: the latter doesn't catch the case when `username` is defined but both `apiKey` and `accessToken` are undefined.